minidump-common/src/format.rs

```diff
@@ -1634,3 +1635,213 @@ pub struct DSO_DEBUG_64 {
     /// The address of the "dynamic structure"
     pub dynamic: u64,
 }
+
+/// A variable-length UTF-8-encoded string carried within a minidump file.
+#[derive(Clone)]
+pub struct MINIDUMP_STRING {
+    /// The length of the #Buffer field in bytes, not including the `NUL` terminator.
+    ///
+    /// This field is interpreted as a byte count, not a count of Unicode code points.
+    pub length: u32,
+    /// The string, encoded in UTF-8, and terminated with a `NUL` byte.
+    pub buffer: Vec<u8>,
+}
+
+impl<'a> scroll::ctx::TryFromCtx<'a, Endian> for MINIDUMP_STRING {
```

Comment:
Agreed. This is also a common pattern in PDBs, where the length field is even offset from the variable-length buffer. It's not too bad to write a manual impl, though :)

Reply:
Also, this reminds me that we ditched `scroll-derive` in pdb for faster compile times. It turned out that scroll takes pretty long to derive, and in many cases the manual derives are easy enough to maintain.

Reply:
That's a bummer! Making the compiler do this work and not having to maintain boilerplate implementations with all the potential hazards that entails is one of the big selling points for scroll for me. Is it just the fact that you don't have any other dependencies that pull in `syn`, so you eat that whole cost for it? This would be a useful data point in a discussion about Rust improving the state of proc macros.

Reply:
Fully agreed. I didn't perform the benchmarks myself but knowing `syn`, it is likely the culprit. The change was made in https://github.com/willglynn/pdb/pull/70.